### PR TITLE
Fill in comparison methods and unit tests for ResourceType

### PR DESCRIPTION
**Pull request by HarveyLink**
ADO Work Item: https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_workitems/edit/5004

### CODE REVIEW

File: Azure.ResourceManager.Core.Tests/ResourceTypeTests.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Core.Tests
+{
+    public class ResourceTypeTests
+    {
+        const string resourcePrefix = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRg/providers/";
+
+        [TestCase(0, "Microsoft.ClassicStorage/storageAccounts/account1",
```
Comment (m-nash) [inline Azure.ResourceManager.Core.Tests/ResourceTypeTests.cs:12]: add a test to compare to null

File: Azure.ResourceManager.Core.Tests/ResourceTypeTests.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Core.Tests
+{
+    public class ResourceTypeTests
```
Comment (m-nash) [inline Azure.ResourceManager.Core.Tests/ResourceTypeTests.cs:8]: Some of the tests should pass in just the resourceType instead of the entire id, we should also compare the combinations.  Right now we are only comparing full id to full id
id.CompareTo(resourceType)
resourceType.CompareTo(id)
resourceType.CompareTo(resourceType)

**Review by m-nash — CHANGES_REQUESTED** [review]

File: Azure.ResourceManager.Core.Tests/ResourceTypeTests.cs (hunk 1 above)
Comment (HarveyLink) [reply Azure.ResourceManager.Core.Tests/ResourceTypeTests.cs]: Since parameter null will throw ArgumentOutOfRangeException exception, I add object null compare here.

File: Azure.ResourceManager.Core/Resources/ResourceType.cs [hunk 3]
```diff
@@ -87,7 +87,7 @@ public ResourceType Parent
 
         public int CompareTo(ResourceType other)
         {
-            if (other == null)
+            if ((object)other == null)
```
Comment (m-nash) [inline Azure.ResourceManager.Core/Resources/ResourceType.cs:90]: typically you use reference equals here instead of a cast
```suggestion
            if (ReferenceEquals(other, null))
```

**Review by m-nash — CHANGES_REQUESTED** [review]

**Review by m-nash — APPROVED** [review]
